analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -129,7 +129,7 @@ token switch
                 };
 
             protected static TokenInfo TokenInfo(SyntaxToken token, TokenType tokenType) =>
-                string.IsNullOrWhiteSpace(token.ValueText)
+                string.IsNullOrWhiteSpace(token.ValueText) && tokenType != TokenType.StringLiteral
```

Comment:
I'm not sure I understand the intention. If we already know the token type why return null? 

This method is now handling only String literals. The method name can be changed to reflect that.